src/components/universal/carousel.js

```diff
@@ -0,0 +1,101 @@
+import React, { Component } from 'react';
+import { Carousel, CarouselItem, CarouselControl, CarouselIndicators, CarouselCaption } from 'reactstrap';
+
+const items = [
+    {
+        src: 'http://placehold.it/1500x300',
+        altText: 'Slide 1',
+        caption: 'Slide 1'
+    },
+    {
+        src: 'http://placehold.it/1500x300',
+        altText: 'Slide 2',
+        caption: 'Slide 2'
+    },
+    {
+        src: 'http://placehold.it/1500x300',
+        altText: 'Slide 3',
+        caption: 'Slide 3'
+    }
+]
+
+class CarouselComponent extends Component {
+    constructor(props) {
+        super(props);
+
+        this.state = { activeIndex: 0 };
+        this.next = this.next.bind(this);
+        this.previous = this.previous.bind(this);
+        this.goToIndex = this.goToIndex.bind(this);
+        this.onExiting = this.onExiting.bind(this);
+        this.onExited = this.onExited.bind(this);
```

Comment:
Define all these methods as arrow functions and you can avoid binding them in the constructor